Modules/_json.c

```diff
@@ -1493,9 +1499,12 @@ encoder_listencode_dict(PyEncoderObject *s, _PyAccu *acc,
                         PyObject *dct, Py_ssize_t indent_level)
 {
     /* Encode Python dict dct a JSON term */
-    static PyObject *open_dict = NULL;
-    static PyObject *close_dict = NULL;
-    static PyObject *empty_dict = NULL;
+    _Py_static_string(PyId_open_dict, "{");
+    _Py_static_string(PyId_close_dict, "}");
+    _Py_static_string(PyId_empty_dict, "{}");
+    PyObject *open_dict = _PyUnicode_FromId(&PyId_open_dict);
+    PyObject *close_dict = _PyUnicode_FromId(&PyId_close_dict);
+    PyObject *empty_dict = _PyUnicode_FromId(&PyId_empty_dict);
```

Comment:
```suggestion
    PyObject *open_dict = _PyUnicode_FromId(&PyId_open_dict);  // borrowed ref
    PyObject *close_dict = _PyUnicode_FromId(&PyId_close_dict);  // borrowed ref
    PyObject *empty_dict = _PyUnicode_FromId(&PyId_empty_dict);  // borrowed ref
```

Same remark for open_array & friends below.